SparkPost/tools-ui
# FE-1396 - Ejected create-react-app tools-ui upgrade

Local: http://localhost:3000/dkim
Stage: https://tools-staging.sparkpost.com/dkim
Prod: https://tools.sparkpost.com/dkim

### What Changed
 - Re-initialized the project with `create-react-app`, then ejected the starter. Then added the src and config files back.
 - Cypress was installed and a few basic tests were added for each of the three pages.
 - The local application no longer redirects to production. [Instead, it redirects to localhost](https://github.com/SparkPost/tools-ui/pull/76/files#diff-481bff25593155cc0695aeb0f112c109e3075ef703d254734ec2a3c5266ee0beR14).  (`npm run build` will still redirect to prod while `npm run start` redirects to localhost)
 - `states/state-max-refresh-retries.json` was deleted.
 - `config/polyfills.js` was deleted.
 - React was upgraded (via create-react-app)
 - WebPack was upgraded (via create-react-app)
 - redux-form was upgraded.

### How to test
 - Run this project locally while having the stage or production app open too.
 - There are three main tools/routes for this app. The DKIM Validator, SPF Inspector, and SPF Builder.
 - We need to make sure that every piece of functionality on all three of those pages/states works exactly as expected and like production. We should be on the lookout for javascript (mostly type) errors especially. 
 - 
 

### Tools-UI Feature List 

#### Navigation
 - Tools icon/button/link goes to the marketing landing page.
 - Login button -> redirects to 2web2ui login (works with local 2web2ui with the right configs)
 - Signup button -> redirects to 2web2ui signup (works with local 2web2ui with the right configs)

#### DKIM Validator Feature List
 - Generate email address
 - View Results takes user to the result view/state (Requires sending an email to the generated address) 
 - **What do we need to send to that email to accurately test the DKIM View Results page?**

#### SPF Inspector Feature List
 - Inspect a domain 
 - View domain inspection results
 - Inspect results: Save Results
 - Inspect results: Share Results
 - Inspect results: Refresh
 - Inspect results: How do I fix errors?
 - Inspect results: Expand / Collapse
 - Inspect results: Shows SPF Record and IP list

#### SPF Builder 
 - Enter a domain
 - MX Records
 - A Records
 - IP Network Ranges
 - Includes
 - How strict should this SPF Record be?
 - Fixed view "Add this TXT record to your DNS" message
 - "check out the inspector" link inside the fixed view message.
 - Copy the generated string/record from adding the different configurations.
